File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Utils/Resources.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using travelAgency2.Repository;
using travelAgency2.Repository.Database;
using travelAgency2.Service;
using travelAgency2.src.Domain;

namespace travelAgency2.Utils
{


    internal class Resources
    {
        static Resources instance = null;
        private static TableFactory tableFactory;
        private UserRepository userRepository;
        private LocationRepository locationRepository;
        private FlightRepository flightRepository;
        private PurchaseRepository purchaseRepository;
        private UserService userService;
        private LoginService loginService;
        private FlightService flightService;
        private PurchaseService purchaseService;
        private Flight selectedFlight = null;
        public static Resources GetInstance()
        {
            if (instance == null)
                instance = new Resources();
            return instance;
        }
        private Resources() 
        {
            tableFactory = new TableFactory();
            userRepository = new UserRepository();
            locationRepository = new LocationRepository();
            flightRepository = new FlightRepository();
            purchaseRepository = new PurchaseRepository();
            userService = new UserService(userRepository);
            loginService = new LoginService();
            flightService = new FlightService(flightRepository);
            purchaseService = new PurchaseService(purchaseRepository, flightService);
        }

        public void SetSelectedFlight(Flight flight)
        {
            this.selectedFlight = flight;
        }

        public Flight GetSelectedFlight()
        {
            return this.selectedFlight;
        }

        public PurchaseService GetPurchaseService()
        {
            return purchaseService;
        }
        public FlightService GetFlightService()
        {
            return flightService;
        }
        public LoginService GetLoginService()
        {
            return loginService;
        }
        public UserService GetUserService()
        {
            return userService;
        }
        public LocationRepository getLocationRepository()
        {
            return locationRepository;
        }
        public static TableFactory getTableFactory()
        {
            return tableFactory;
        }

        public UserRepository getUserRepository()
        {
            return userRepository;
        }

        public FlightRepository getFlightRepository()
        {
            return flightRepository;
        }

        public PurchaseRepository getPurchaseRepository()
        {
            return purchaseRepository;
        }
    }
}

File: second-year/first-semester/MAP/SocialApp/src/main/java/com/example/socialapp/utils/Resources.java
package com.example.socialapp.utils;

import com.example.socialapp.repository.FriendshipRepository;
import com.example.socialapp.repository.UserRepository;
import com.example.socialapp.service.FriendshipService;
import com.example.socialapp.service.UserService;

import java.sql.SQLException;

public final class Resources {
    public static Resources instance = null;
    public static UserService userService;
    public static UserRepository userRepo;
    public static FriendshipRepository friendshipRepo;
    public static FriendshipService friendshipService;

    private Resources() throws SQLException {
        userRepo = new UserRepository();
        friendshipRepo = new FriendshipRepository();
        userService = new UserService(userRepo);
    }


    public static Resources getInstance() throws SQLException {
        if (instance == null) {
            instance = new Resources();
        }
        return instance;
    }
}
